plugins/video-intelligence-monetization/frontend/public/javascripts/countly.views.js

```diff
@@ -0,0 +1,179 @@
+        self.template = Handlebars.compile(src);
+      }),
+      $.get(countlyGlobal["path"] + '/video-intelligence-monetization/templates/bignumbers.html', function(src) {
+        self.bigNumbersTemplate = Handlebars.compile(src);
+      }),
+      countlyMonetization.initialize()
+    );
+  },
+  initializeEventSelector: function() {
+    var self = this;
+    $("#big-numbers-container").on("click", ".check", function() {
+      var classes = $(this).attr('class');
+      var id = $(this).attr('data-id');
+
+      if (classes.indexOf('selected') >= 0) {
+        if (countlyMonetization.tryDisableEvent(id)) {
+          $(this).removeClass("selected");
+        }
+      } else {
+        $(this).addClass("selected");
+        countlyMonetization.enableEvent(id)
+      }
+
+      self.updateChart();
+    });
+  },
+  renderCommon: function(isRefresh) {
+    var self = this;
+    if (!isRefresh) {
+      $(this.el).html(self.template(this.templateData));
+      self.initializeEventSelector()
+      self.updateView(false)
+    }
+  },
+  updateChart: function(fetchData) {
+    var self = this;
+    var data = self.getData();
+    self.renderChart(data.chartDP);
+  },
+  updateView: function(isRefresh) {
+    var self = this;
+    var data = self.getData();
+    self.renderChart(data.chartDP);
+    self.renderNumbers(data.bigNumbersData);
+    self.renderTable(data.tableData, isRefresh);
+  },
+  refresh: function() {
+    var self = this;
+    $.when(
+      countlyMonetization.requestMetricData()
+    ).done(function() {
+      self.updateView(true)
+    });
+  },
+  renderChart: function(data) {
+    var enabledLines = countlyMonetization.getEnabledEvents();
+    var dataToBeRendered = [];
+    enabledLines.forEach(function(key) {
+      dataToBeRendered.push(data[key]);
+    })
+    return countlyCommon.drawTimeGraph(dataToBeRendered, "#chartContainer");
+  },
+  renderNumbers: function(data) {
+    var self = this;
+    var compiled = self.bigNumbersTemplate({
+      "bignumbers": data
+    });
+
+    $("#big-numbers-container").html(compiled);
+  },
+  renderTable: function(data, isRefresh) {
+    var self = this;
+
+    var tableData = data
+    if (isRefresh) {
+      CountlyHelpers.refreshTable($('#dataTable').dataTable(), tableData);
+    } else {
+      self.createTable(data);
+    }
+  },
+  createTable: function(data, destroy) {
+    var aoColumns = [{
+      "mData": "date",
+      "sType": "customDate",
+      "sTitle": jQuery.i18n.map["common.date"]
+    }];
+
+    var eventColumns = countlyMonetization.getColumns();
+
+    for (var ekey in eventColumns) {
+      aoColumns.push({
+        "mData": ekey,
+        sType: "formatted-num",
+        "mRender": function(d) {
+          return countlyCommon.formatNumber(d);
+        },
+        "sTitle": jQuery.i18n.map[eventColumns[ekey].localize]
+      }, )
+    }
+
+    if (destroy && $.fn.dataTable.fnIsDataTable('#dataTable')) {
+      $("#dataTable").dataTable().fnDestroy();
```

Comment:
I found that, this part of code never works, since columns are fixed. Also, createTable is only called once, to literally create the table. So, it is not possible to recreate it. In refresh case, I already use refresh helper.

I remove that 'destroy' part. If it is needed, then we can refactor.